tests/run/sys_monitoring.py

```diff
@@ -0,0 +1,593 @@
+    f_withgil_prof PY_RETURN [10], LINE [10]
+    m_cdef PY_RETURN [10], LINE [10]
+    m_cpdef PY_RETURN [20], LINE [20]
+    m_def PY_RETURN [20], LINE [20]
+    test_profile PY_RETURN [1], LINE [314], RAISE [20]
+    --- PY_START, PY_RETURN, LINE, RAISE ---
+    720
+    f_cdef PY_START [10], PY_RETURN [10], LINE [10]
+    f_cpdef PY_START [20], PY_RETURN [20], LINE [20]
+    f_def PY_START [10], PY_RETURN [10], LINE [10]
+    f_inline PY_START [10], PY_RETURN [10], LINE [10]
+    f_inline_prof PY_START [10], PY_RETURN [10], LINE [10]
+    f_nogil_prof PY_START [10], PY_RETURN [10], LINE [10]
+    f_raise PY_START [20], LINE [20], RAISE [20], PY_UNWIND [20]
+    f_reraise PY_START [10], LINE [50], RAISE [10], PY_UNWIND [10], RERAISE [10]
+    f_return_default PY_START [10], PY_RETURN [10], LINE [10]
+    f_return_none PY_START [10], PY_RETURN [10], LINE [10]
+    f_withgil_prof PY_START [10], PY_RETURN [10], LINE [10]
+    m_cdef PY_START [10], PY_RETURN [10], LINE [10]
+    m_cpdef PY_START [20], PY_RETURN [20], LINE [20]
+    m_def PY_START [20], PY_RETURN [20], LINE [20]
+    test_profile PY_START [1], PY_RETURN [1], LINE [314], RAISE [20]
+
+
+    ## Testing fused functions:
+
+    >>> with monitored_events(events=FUNC_EVENTS, function_name='call_fused_functions') as collected_events:
+    ...     call_fused_functions()
+    >>> print_events(collected_events)  # call_fused_functions()
+    call_fused_functions PY_START [1], PY_RETURN [1]
+
+
+    >>> smon.free_tool_id(TOOL_ID)
+    """
+
+
+def names(event_ids):
+    return ', '.join([event_name(1 << event_id) for event_id in sorted(event_ids)])
+
+
+def print_events(collected_events):
+    def event_sum(event_id, event_counts):
+        count = sum(event_counts)
+        if not cython.compiled and event_id == E.RERAISE and count == 20:
+            # CPython bug, see https://github.com/python/cpython/issues/118360
+            count //= 2
+        return count
+
+    for func_name, line_counts in sorted(collected_events.items()):
+        print(func_name,
+            ', '.join(f"{names([event_id])} [{event_sum(event_id, counts.values())}]" for event_id, counts in sorted(line_counts.items())))
+
+
+def assert_events(expected_events, collected_events, loops=10):
+    missing = set(expected_events).difference(collected_events)
+    surplus = set(collected_events).difference(expected_events)
+
+    if missing:
+        print(f"Expected: [{names(expected_events)}], missing: [{names(missing)}]")
+    if surplus:
+        print(f"Expected: [{names(expected_events)}], surplus: [{names(surplus)}]")
+
+    collected_events = {name: dict(values) for name, values in collected_events.items()}
+
+    return
+    if cython.compiled:
+        if E.PY_START in collected_events:
+            print("START", collected_events[E.PY_START])
+        if E.PY_RETURN in collected_events:
+            print("RETURN", collected_events[E.PY_RETURN])
+        print(collected_events)
```

Comment:
Obviously not run (may be deliberate though...)

Reply:
Nothing valuable, though. I'll remove it.